pydantic/_internal/_decorators.py

```diff
@@ -218,3 +256,127 @@ def in_ipython() -> bool:
         return False
     else:  # pragma: no cover
         return True
+
+
+class OnlyValueValidator(Protocol):
+    def __call__(self, __value: Any) -> Any:
+        ...
+
+
+class V1ValidatorWithValues(Protocol):
+    def __call__(self, __value: Any, values: dict[str, Any]) -> Any:
+        ...
+
+
+class V1ValidatorWithValuesKwOnly(Protocol):
+    def __call__(self, __value: Any, *, values: dict[str, Any]) -> Any:
+        ...
+
+
+class V1ValidatorWithKwargs(Protocol):
+    def __call__(self, __value: Any, **kwargs: Any) -> Any:
+        ...
+
+
+class V1ValidatorWithKwargsAndValue(Protocol):
+    def __call__(self, __value: Any, values: dict[str, Any], **kwargs: Any) -> Any:
+        ...
+
+
+class V1ValidatorWithValuesAndKwargs(Protocol):
+    def __call__(self, __value: Any, values: dict[str, Any], **kwargs: Any) -> Any:
+        ...
+
+
+V1Validator = Union[
+    V1ValidatorWithValues, V1ValidatorWithValuesKwOnly, V1ValidatorWithKwargs, V1ValidatorWithValuesAndKwargs
+]
+
+
+def make_generic_validator(
+    validator: V1Validator | OnlyValueValidator | ValidatorFunction, mode: str
+) -> ValidatorFunction:
+    """
+    Make a generic function which calls a validator with the right arguments.
+    """
+    sig = signature(validator)
+
+    def _warn_v1_validator() -> None:
+        warnings.warn(
+            'Validator signatures using the `values` keyword argument or `**kwargs` are no longer supported.'
```

Comment:
```suggestion
            'One of the validators in this class has a signature that uses the `values` keyword argument or `**kwargs`, which are no longer supported.'
```

This is perhaps splitting hairs a bit, but I'm thinking that if (according to the note below) the warning will just show that it is coming from the class, and not a specific line, it might be worth the extra bit of clarity in this warning.

Actually — is it the case that this warning can only be produced in the context of defining a class? If so, I think my suggestion might be sensible, but if it can be produced in other contexts it might be worth confirming that the `stacklevel` keyword argument is still doing something reasonable.

Reply:
I think it might show up as well if you do `create_model`. In that case it would look like it's coming from within `create_model`, which I think is fine enough...

Reply:
For what it's worth, my understanding was that the plan was to get rid of `create_model`. Not 100% sure about that though.

Reply:
Even if we don't get rid of it I'd still consider it a somewhat advanced use case and at that point you should be able to figure out that the warning is caused by you 1 stack frame up.